zero_bin/leader/src/client.rs

```diff
@@ -131,3 +145,29 @@ pub(crate) async fn client_main(
 
     Ok(())
 }
+
+fn check_previous_proof(
+    checkpoint_block_number: u64,
+    previous_proof: Option<GeneratedBlockProof>,
+    start: u64,
+) -> Result<()> {
+    if previous_proof.is_some() {
+        if previous_proof.clone().unwrap().b_height + 1 != start {
```

Comment:
Don't unwrap in production code. If error happens and you can not continue, return error from function.